cmd/ipfs-cluster-service/daemon.go

```diff
@@ -98,25 +90,23 @@ func daemon(c *cli.Context) error {
 	// will realize).
 	go bootstrap(ctx, cluster, bootstraps)
 
-	return handleSignals(ctx, cluster)
+	return handleSignals(ctx, cluster, store)
 }
 
 func createCluster(
 	ctx context.Context,
 	c *cli.Context,
 	cfgs *cfgs,
 	raftStaging bool,
-) (*ipfscluster.Cluster, error) {
-	err := observations.SetupMetrics(cfgs.metricsCfg)
-	checkErr("setting up Metrics", err)
-
-	tracer, err := observations.SetupTracing(cfgs.tracingCfg)
-	checkErr("setting up Tracing", err)
+) (*ipfscluster.Cluster, ds.Datastore, error) {
```

Comment:
Because we need to close the datastore after we shutdown the Cluster. Another possibility is
passing it to the main component and letting it Close it on shutdown (but that's all the main component would do with it). Would this be better? I'm leaning to say yes...

Reply:
I would prefer passing it to the main component as well. I think the main component should manage the lifecycle of all the cluster components/dependencies.

Reply:
@hsanjuan this is the only thing I would like to see changed before approving.